weaver/samples/fabric/satpsimpleasset/assetmgmt_test.go

```diff
@@ -0,0 +1,232 @@
+package main_test
+
+import (
+	"encoding/json"
+	"encoding/base64"
+	"crypto/sha256"
+	"fmt"
+	"testing"
+	"time"
+
+	mspProtobuf "github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/stretchr/testify/require"
+	sa "github.com/hyperledger/cacti/weaver/samples/fabric/satpsimpleasset"
+	"github.com/hyperledger/cacti/weaver/common/protos-go/v2/common"
+	wtest "github.com/hyperledger/cacti/weaver/core/network/fabric-interop-cc/libs/testutils"
+)
+
+// function that supplies value that is to be returned by ctx.GetStub().GetCreator() in locker/recipient context
+func getCreatorInContext(creator string) string {
+	serializedIdentity := &mspProtobuf.SerializedIdentity{}
+	var eCertBytes []byte
+	if creator == "locker" {
+		eCertBytes, _ = base64.StdEncoding.DecodeString(getLockerECertBase64())
+	} else {
+		eCertBytes, _ = base64.StdEncoding.DecodeString(getRecipientECertBase64())
+	}
+	serializedIdentity.IdBytes = eCertBytes
+	serializedIdentity.Mspid = "ca.org1.example.com"
+	serializedIdentityBytes, _ := proto.Marshal(serializedIdentity)
+
+	return string(serializedIdentityBytes)
+}
+
+// function that supplies the ECert in base64 for locker (e.g., Alice)
+func getLockerECertBase64() string {
+	eCertBase64 := "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNyVENDQWxTZ0F3SUJBZ0lVSENXLzBtV0xhc2hISG9zd0xxVWhpK1FwREc4d0NnWUlLb1pJemowRUF3SXcKY2pFTE1Ba0dBMVVFQmhNQ1ZWTXhGekFWQmdOVkJBZ1REazV2Y25Sb0lFTmhjbTlzYVc1aE1ROHdEUVlEVlFRSApFd1pFZFhKb1lXMHhHakFZQmdOVkJBb1RFVzl5WnpFdWJtVjBkMjl5YXpFdVkyOXRNUjB3R3dZRFZRUURFeFJqCllTNXZjbWN4TG01bGRIZHZjbXN4TG1OdmJUQWVGdzB5TURBM01qa3dORE0yTURCYUZ3MHlNVEEzTWprd05EUXgKTURCYU1GMHhDekFKQmdOVkJBWVRBbFZUTVJjd0ZRWURWUVFJRXc1T2IzSjBhQ0JEWVhKdmJHbHVZVEVVTUJJRwpBMVVFQ2hNTFNIbHdaWEpzWldSblpYSXhEekFOQmdOVkJBc1RCbU5zYVdWdWRERU9NQXdHQTFVRUF4TUZkWE5sCmNqRXdXVEFUQmdjcWhrak9QUUlCQmdncWhrak9QUU1CQndOQ0FBU3VoL3JWQ2Y4T0R1dzBJaG5yTTJpaWYyYTcKc0dUOEJJVjFQRURVM1NucUNsbWgrUlYvM0p5S2wvVHl0aHpOL1pWbktFL3R2NWQzZ1ZXYk5zdGM5NytTbzRIYwpNSUhaTUE0R0ExVWREd0VCL3dRRUF3SUhnREFNQmdOVkhSTUJBZjhFQWpBQU1CMEdBMVVkRGdRV0JCUXgvaExZCkNORzRlekNxdmdUS0MvV3d1U1ZubURBZkJnTlZIU01FR0RBV2dCVFdENjArZUNIYkR5RDMzUFdiQ3hWdVFxTUEKcVRBZkJnTlZIUkVFR0RBV2doUnZZelV4TURNM05EY3pPREF1YVdKdExtTnZiVEJZQmdncUF3UUZCZ2NJQVFSTQpleUpoZEhSeWN5STZleUpvWmk1QlptWnBiR2xoZEdsdmJpSTZJaUlzSW1obUxrVnVjbTlzYkcxbGJuUkpSQ0k2CkluVnpaWEl4SWl3aWFHWXVWSGx3WlNJNkltTnNhV1Z1ZENKOWZUQUtCZ2dxaGtqT1BRUURBZ05IQURCRUFpQUYKbnNMNlV1eFRtSks5bmhkTU1QNWxWN3hueVlsMVd5RGl6RVFzZnd1T1p3SWdYY3duSE9hVURXWWpmWHRGU0k1eQp6WjltcjZQRWtSNER0VEhJUkZhTVYxOD0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQ=="
```

Comment:
Can you generate the certificates instead of hardcoding them? Since it's in a test this is not critical, but nevertheless recommended

Reply:
@outSH I agree with your suggestion. For now, let's leave it is. This kind of hardcoded serialized certificate is present in several unit tests, and we can refactor everything in a separate PR.